include/zephyr/drivers/misc/ft8xx/ft8xx.h

```diff
@@ -45,8 +46,11 @@ struct ft8xx_touch_transform {
  * @brief Callback API to inform API user that FT8xx triggered interrupt
  *
  * This callback is called from IRQ context.
+ *
+ * @param dev Pointer to the device structure for the driver instance
+ * @param user_data Pointer to user data provided during callback registration
  */
-typedef void (*ft8xx_int_callback)(void);
+typedef void (*ft8xx_int_callback)(const struct device *dev, void *user_data);
```

Comment:
adding support for user data seems to be unrelated to adding support for multiple instances. It would be better to split the changes in different commits